File: hahlw/array/27.移除元素.cpp
/*
 * @lc app=leetcode.cn id=27 lang=cpp
 *
 * [27] 移除元素
 */
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int cnt=0;
        for(int i=0;i<nums.size();i++){
            nums[i-cnt]=nums[i];
            if(nums[i]==val)cnt++;
        }
        // for(int i = 0;i<nums.size();i++){
        //     cout<<nums[i];
        // }
        return nums.size()-cnt;
    }
};

File: Juniper/27.移除元素.py
#
# @lc app=leetcode.cn id=27 lang=python3
#
# [27] 移除元素
#
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i = 0
        lens = len(nums)
        while i < lens:
            if nums[i]== val:
                del nums[i]
                lens -=1 
            else :
                i +=1
        return lens
            

        

